corehq/form_processor/tests/test_models.py

```diff
@@ -129,3 +131,54 @@ def test_case_indices(self):
 def are_cases_equal(a, b):  # or at least equal enough for our test
     attrs = ('domain', 'case_id', 'type', 'name', 'owner_id')
     return all(getattr(a, attr) == getattr(b, attr) for attr in attrs)
+
+
+def test_case_to_json():
```

Comment:
👍🏻 thanks for this test - we were just having a conversation about this behaviour yesterday (user defined properties with the same names as system properties)